cli/cmd/root.go

```diff
@@ -274,6 +289,45 @@ func registryOverride(image, registry string) string {
 	return strings.Replace(image, defaultDockerRegistry, registry, 1)
 }
 
+// mapRangeSlice creates and populates the target slice of unsigned integers
+// based upon the source slice of string integer(s) and range(s).
+// For example:
+//   []string{"23","25-27"}
+// will result in
+//   []uint{23,25,26,27}
+func mapRangeSlice(target *[]uint, source []string) error {
+	*target = make([]uint, 0)
+	for _, portOrRange := range source {
+		if strings.Contains(portOrRange, "-") {
+			bounds := strings.Split(portOrRange, "-")
+			if len(bounds) != 2 {
+				return fmt.Errorf("ranges expected as <lower>-<upper>")
+			}
+			lower, err := strconv.ParseUint(bounds[0], 10, 0)
+			if err != nil {
+				return fmt.Errorf("\"%s\" is not a valid lower-bound", bounds[0])
+			}
+			upper, err := strconv.ParseUint(bounds[1], 10, 0)
+			if err != nil {
+				return fmt.Errorf("\"%s\" is not a valid upper-bound", bounds[1])
+			}
+			if upper < lower {
+				return fmt.Errorf("\"%s\": upper-bound must be greater than lower-bound", portOrRange)
+			}
+			for i := lower; i <= upper; i++ {
+				*target = append(*target, uint(i))
```

Comment:
+1 For testing the bounds

Reply:
Thanks!  This also made me realize that I did not include a test for the upper- and lower-bounds being equal.